### ISSUE TITLE
openspliceai. ERROR - Error processing model for device

### ISSUE THREAD

**Issue by morganc6** [Kuanhao-Chao/OpenSpliceAI#8]
Hello there,

I am trying to run openspliceai variant using the precomputed openspliceai-mane models. I dont know why but it seems unable to process the models that are available. Here is the code

```
openspliceai variant \
--input-vcf  sample.vcf" \
--ref-genome Homo_sapiens_assembly38.fasta \
--annotation grch38.txt \
--model ./models/openspliceai-mane/10000nt \
--model-type pytorch\
--flanking-size 400 \
--distance 100 \
--mask 1 \
--output-vcf sample.openspliceai.vcf

```
and the error message is as follows:
2025-04-10 16:33:17,256 - INFO - Initializing Annotator class
2025-04-10 16:33:17,717 - ERROR - Error processing model for device: Error(s) in loading state_dict for SpliceAI:
        Unexpected key(s) in state_dict: "residual_units.10.batchnorm1.weight", "residual_units.10.batchnorm1.bias", "residual_units.10.batchnorm1.running_mean", "residual_units.10.batchnorm1.running_var", "residual_units.10.batchnorm1.num_batches_tracked", "residual_units.10.batchnorm2.weight", "residual_units.10.batchnorm2.bias", "residual_units.10.batchnorm2.running_mean", "residual_units.10.batchnorm2.running_var", "residual_units.10.batchnorm2.num_batches_tracked", "residual_units.10.conv1.weight", "residual_units.10.conv1.bias", "residual_units.10.conv2.weight", "residual_units.10.conv2.bias", "residual_units.11.batchnorm1.weight", "residual_units.11.batchnorm1.bias", "residual_units.11.batchnorm1.running_mean", "residual_units.11.batchnorm1.running_var", "residual_units.11.batchnorm1.num_batches_tracked", "residual_units.11.batchnorm2.weight", "residual_units.11.batchnorm2.bias", "residual_units.11.batchnorm2.running_mean", "residual_units.11.batchnorm2.running_var", "residual_units.11.batchnorm2.num_batches_tracked", "residual_units.11.conv1.weight", "residual_units.11.conv1.bias", "residual_units.11.conv2.weight", "residual_units.11.conv2.bias", "residual_units.12.batchnorm1.weight", "residual_units.12.batchnorm1.bias", "residual_units.12.batchnorm1.running_mean", "residual_units.12.batchnorm1.running_var", "residual_units.12.batchnorm1.num_batches_tracked", "residual_units.12.batchnorm2.weight", "residual_units.12.batchnorm2.bias", "residual_units.12.batchnorm2.running_mean", "residual_units.12.batchnorm2.running_var", "residual_units.12.batchnorm2.num_batches_tracked", "residual_units.12.conv1.weight", "residual_units.12.conv1.bias", "residual_units.12.conv2.weight", "residual_units.12.conv2.bias", "residual_units.13.batchnorm1.weight", "residual_units.13.batchnorm1.bias", "residual_units.13.batchnorm1.running_mean", "residual_units.13.batchnorm1.running_var", "residual_units.13.batchnorm1.num_batches_tracked", "residual_units.13.batchnorm2.weight", "residual_units.13.batchnorm2.bias", "residual_units.13.batchnorm2.running_mean", "residual_units.13.batchnorm2.running_var", "residual_units.13.batchnorm2.num_batches_tracked", "residual_units.13.conv1.weight", "residual_units.13.conv1.bias", "residual_units.13.conv2.weight", "residual_units.13.conv2.bias", "residual_units.14.conv.weight", "residual_units.14.conv.bias", "residual_units.15.batchnorm1.weight", "residual_units.15.batchnorm1.bias", "residual_units.15.batchnorm1.running_mean", "residual_units.15.batchnorm1.running_var", "residual_units.15.batchnorm1.num_batches_tracked", "residual_units.15.batchnorm2.weight", "residual_units.15.batchnorm2.bias", "residual_units.15.batchnorm2.running_mean", "residual_units.15.batchnorm2.running_var", "residual_units.15.batchnorm2.num_batches_tracked", "residual_units.15.conv1.weight", "residual_units.15.conv1.bias", "residual_units.15.conv2.weight", "residual_units.15.conv2.bias", "residual_units.16.batchnorm1.weight", "residual_units.16.batchnorm1.bias", "residual_units.16.batchnorm1.running_mean", "residual_units.16.batchnorm1.running_var", "residual_units.16.batchnorm1.num_batches_tracked", "residual_units.16.batchnorm2.weight", "residual_units.16.batchnorm2.bias", "residual_units.16.batchnorm2.running_mean", "residual_units.16.batchnorm2.running_var", "residual_units.16.batchnorm2.num_batches_tracked", "residual_units.16.conv1.weight", "residual_units.16.conv1.bias", "residual_units.16.conv2.weight", "residual_units.16.conv2.bias", "residual_units.17.batchnorm1.weight", "residual_units.17.batchnorm1.bias", "residual_units.17.batchnorm1.running_mean", "residual_units.17.batchnorm1.running_var", "residual_units.17.batchnorm1.num_batches_tracked", "residual_units.17.batchnorm2.weight", "residual_units.17.batchnorm2.bias", "residual_units.17.batchnorm2.running_mean", "residual_units.17.batchnorm2.running_var", "residual_units.17.batchnorm2.num_batches_tracked", "residual_units.17.conv1.weight", "residual_units.17.conv1.bias", "residual_units.17.conv2.weight", "residual_units.17.conv2.bias", "residual_units.18.batchnorm1.weight", "residual_units.18.batchnorm1.bias", "residual_units.18.batchnorm1.running_mean", "residual_units.18.batchnorm1.running_var", "residual_units.18.batchnorm1.num_batches_tracked", "residual_units.18.batchnorm2.weight", "residual_units.18.batchnorm2.bias", "residual_units.18.batchnorm2.running_mean", "residual_units.18.batchnorm2.running_var", "residual_units.18.batchnorm2.num_batches_tracked", "residual_units.18.conv1.weight", "residual_units.18.conv1.bias", "residual_units.18.conv2.weight", "residual_units.18.conv2.bias", "residual_units.19.conv.weight", "residual_units.19.conv.bias". . Skipping...
2025-04-10 16:33:17,725 - ERROR - Error processing model for device: Error(s) in loading state_dict for SpliceAI:

Any thoughts or insights would be greatly appreciated!

Thank you,
Claire

**Comment by kasia-te** [comment]
Hi,
Try using the same flanking length that was used during model training (-f 10000). That works for me.
Best,
Kasia

**Comment by conbward** [comment]
Thanks for the suggestion. Another workaround we found was to edit the base code for the variant function in openspliceai/variant/utils.py at line 117 which the state_dicts are loaded and adding strict=False.

`model.load_state_dict(state_dict, strict=False)`

I believe other subcommands had explicit code to handle unnecessary keys (for example openspliceai/calibrate/model_utils.py lines 49-62), and the variant subcommand may benefit from this as well.
